bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java

```diff
@@ -595,4 +597,116 @@ private boolean getConfigParameter(Map<String, Object> parameters, String parame
+     * This list can for example, be used to scan the network for available devices.
+     * 
+     * @return A full list of IP {@link InetAddress} (except network and broadcast)
+     */
+    public static List<InetAddress> getFullRangeOfAddressesToScan() {
+        List<InetAddress> addressesToScan = List.of();
+        List<CidrAddress> ipV4InterfaceAddresses = NetUtil.getAllInterfaceAddresses().stream()
+                .filter(a -> a.getAddress() instanceof Inet4Address).collect(Collectors.toList());
+
+        for (CidrAddress i : ipV4InterfaceAddresses) {
+            addressesToScan.addAll(getAddressesRangeByCidrAddress(i, i.getPrefix()));
+        }
+        return addressesToScan;
+    }
+
+    /**
+     * For the given {@link CidrAddress} all Ipv4 addresses are returned.
+     * This list can for example, be used to scan the network for available devices.
+     * 
+     * @param iFaceAddress The {@link CidrAddress} of the network interface
+     * @param maxPrefixLength Control the prefix length of the network (e.g. 24 for class C)
+     * @return A full list of IP {@link InetAddress} (except network and broadcast)
+     */
+    public static List<InetAddress> getAddressesRangeByCidrAddress(CidrAddress iFaceAddress, int maxPrefixLength) {
+        if (!(iFaceAddress.getAddress() instanceof Inet4Address) || iFaceAddress.getPrefix() < maxPrefixLength) {
+            return List.of();
+        }
+
+        List<byte[]> addresses = getAddressesInSubnet(iFaceAddress.getAddress().getAddress(), iFaceAddress.getPrefix(),
+                true);
+        if (addresses.size() > 2) {
+            addresses.remove(0); // remove network address
+            addresses.remove(addresses.size() - 1); // remove broadcast address
+        }
+
+        return addresses.stream().map(m -> {
+            try {
+                return InetAddress.getByAddress(m);
+            } catch (UnknownHostException e) {
+                return null;
+            }
+        }).filter(f -> f != null).sorted((a, b) -> {
+            byte[] aOct = a.getAddress();
+            byte[] bOct = b.getAddress();
+            int r = 0;
+            for (int i = 0; i < aOct.length && i < bOct.length; i++) {
+                r = Integer.compare(aOct[i] & 0xff, bOct[i] & 0xff);
+                if (r != 0) {
+                    return r;
+                }
+            }
+            return r;
+        }).collect(Collectors.toList());
+    }
+
+    /**
+     * Recursively calculate each IP address within a subnet
+     * 
+     * @param address IP address in byte array form (i.e. 127.0.0.1 = 01111111 00000000 00000000 00000001)
+     * @param maskLength Network mask length (i.e. the number after the forward-slash, '/', in CIDR notation)
+     * @param scrub Whether or not to scrub the unmasked bits of the address
+     * @return All possible IP addresses
+     * 
+     * @author: https://gist.github.com/ssttevee/b0e2b431f4b23d289537
```

Comment:
Implementation could be simpler in our case, as we always scrub and as addresses are always IPv4.
Then it could be like
- calculate the netmask from the prefix length
- calculate lowest address, i.e. address&netmask (this is the scrub)
- calculate upper address, i.e. address&netmask | ~netmask
- loop

Reply:
I changed the implementation so the gist is no longer there. Thanks, the suggestion reduced and simplified it dramatically.